crates/typst-layout/src/shapes.rs

```diff
@@ -365,6 +365,9 @@ fn layout_shape(
         let default = Size::new(Abs::pt(45.0), Abs::pt(30.0));
         let mut size = region.expand.select(region.size, default.min(region.size));
         if kind.is_quadratic() {
+            if region.expand.any(|ex| *ex) {
+                size = region.size;
+            }
```

Comment:
I must admit I'm not too familiar with the surrounding code, but if I understood correctly, the problem is that `expand` will be false for the missing dimension (e.g. `#square(height: 5cm)` is missing width, which will be `auto`, so `expand` will be false on the horizontal axis), in which case, from the `select` above, the height will be kept in `size`, but the width will be set to `default.min(region.size)`, which will be smaller than the height (unless it is smaller than the default width), and thus `size.min_by_side()` will assign the default width as the height as well, leading to the bug. The buggy behavior here (at least conceptually) is precisely that the specified height should actually take priority over the unspecified width, thus it should not be determining the minimum size from both the specified region and the default, only from the specified region. (Unless no dimensions were specified, in which case the default should be used.) Does this check out?

To be honest, it feels a bit unclear to me why `expand` and `min` are being used to detect auto dimensions (where `min` will always return the default when region.size is infinity / auto); in previous code, there was an explicit `unwrap` for `auto` dimensions. I feel like this could be hiding some other bug.

Either way, I think there should be some more comments in the code to make this more explicit, in particular:

1. Indicating that `default.min(region.size)` will return the default dimensions when the region size is infinite (`auto`).
2. Indicating that `size.min_by_side()` should only consider the specified region size when at least one region dimension is fixed (in which case `expand` will be true), ignoring the other dimension set to the default value, if it received that value, to give priority to the specified dimension. This is, in theory, only necessary to check for quadratic shapes as they must be resized to ensure they have equal dimensions. Only the resizing is a problem.
(Though I do wonder if there could be some edge case generating problems with non-quadratic shapes as well, but I could not find one.)

Reply:
> To be honest, it feels a bit unclear to me why expand and min are being used to detect auto dimensions (where min will always return the default when region.size is infinity / auto); in previous code, there was an explicit unwrap for auto dimensions. I feel like this could be hiding some other bug.

Because the sizing now comes from the block around it (which enables fr-sizing). So expand is the only place where the information can come from.

Reply:
I see. I assume it's impossible for expand to mean anything else here then?